packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -157,6 +164,23 @@ abstract class ProjectBase extends Resource implements IProject {
   /** The IAM service Role of this Project. */
   public abstract readonly role?: iam.IRole;
 
+  /**
+   * Defines a CloudWatch event rule triggered when something happens with this project.
+   *
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/sample-build-notifications.html
+   */
+  public onEvent(id: string, options: events.OnEventOptions): events.Rule {
```

Comment:
I guess I missed this in the previous PR, but it's now or never: I kind of think that "target" shouldn't be under "Options", so I can do `onEvent('foo', target)`. 

It also maps more cleanly to the fact that you pass in `options` to `events.Rule` and then `addTarget` separately.

I would also say that `events.RuleProps` should contain `targets` (plural) to allow adding targets declaratively...

Reply:
I was bivaricating on that. I found the API very ugly that way.

Hmm.

Reply:
Actually, here's why I changed the signature: the signature with 3 arguments becomes very ugly if you have a complex target (such as an ECS task or CodeBuild project), or a complex payload (from a JSON message), *and then* also configuration after it.

It would look something like this:

```ts
source.onEvent('MyEvent', rule.addTarget(new targets.EcsEc2Task({
  cluster,
  taskDefinition,
  taskCount: 1,
  containerOverrides: [{
    containerName: 'TheContainer',
    environment: [{
      name: 'I_WAS_TRIGGERED',
      value: 'From CloudWatch Events'
    }]
  }]
}), {
  eventDescription: 'hello'
  eventPattern: {
    field: ['value']
  }
});
```

vs:

```ts
source.onEvent('MyEvent', {
  eventDescription: 'hello'
  eventPattern: {
    field: ['value']
  },
  target: new targets.EcsEc2Task({
    cluster,
    taskDefinition,
    taskCount: 1,
    containerOverrides: [{
        containerName: 'TheContainer',
        environment: [{
        name: 'I_WAS_TRIGGERED',
        value: 'From CloudWatch Events'
        }]
    }]
  })
});
```

I feel the bad case is a lot worse for the first API, where the 2nd API fixes that, and the regular case isn't THAT much worse.

Reply:
Also, I think `obj.method('id', { ...props... })` is more consistent with most of our other APIs.